### ISSUE TITLE
What happens ?

### ISSUE THREAD

**Issue by ghost** [smallnest/go-web-framework-benchmark#9]
Before the last iris was first with fasthttp-raw,
before 18 days, as this commit also says:
https://github.com/smallnest/go-web-framework-benchmark/commit/0874ce64aa6d78235c9452bfed0b8fb6fb4f4547.

Also I didn't change any performance-releated with `SetBody & router`, I just tested with a version we had before a month and the benchmark results are the same as the latest iris version.

Can you tell me when you upgraded iris? what version did you had before (or a date) you did several tests within this period, this is the first time iris is down.

And if you didn't changed anything(iris-releated) from the previous week, could you please re-run the tests?

**Edit2**:

I just ran the benchmarks from docker with `docker pull smallnest/go-web-framework-benchmark`  `docker run  -v /opt/data:/data smallnest/go-web-framework-benchmark` .

Iris is still side-by-side with fasthttp. Some other frameworks I saw on this repo's README that seems to be top of iris, in real, they have lowest results...

For now, only the basic tests (0ms 5000 clients) are finished:

[![fasthttp and others](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/fasthttp.png)](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/fasthttp.png)

[![iris](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/iris.png)](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/iris.png)

I ran it from a 'clean/fresh' docker*

but on README's Basic tests are:

[![README page](https://github.com/smallnest/go-web-framework-benchmark/raw/master/benchmark.png)](https://github.com/smallnest/go-web-framework-benchmark/raw/master/benchmark.png)

How is that even possible?

**Edit 3**: 10ms 5k clients finished also:

[![fasthttp and others 10ms](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/fasthttp10ms.png)](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/fasthttp10ms.png)

[![iris 10ms](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/iris10ms.png)](https://github.com/iris-contrib/website/raw/gh-pages/assets/docker_results/iris10ms.png)

I will leave the machine open until the graphical results also... I really need to sleep now

But you got the idea.. something happened on your last machine's benchmark testing

Thanks for everything!!

**Comment by smallnest** [comment]
I remove iris and  run `go get -u`.
I started a new test.

**Comment by smallnest** [comment]
Iris still is No. 1.
I update test results (pipeline has not updated yet because test has not been finished, I will update all test results tomorrow).

I think the reason is I have not updated latest web frameworks.

**Comment by razonyang** [comment]
But i found that clevergo is missing, although it is not important for me, hope to add it next time. :) @smallnest

@headwindfly I don't know why clevergo has not been included in test results. Codes exist here. I will re-run the test and check the test result. Thanks.

**Comment by kataras** [comment]
@smallnest 

Thanks you man, you understand the shock I had, thanks for the update!

also I just saw some 'new' frameworks come up like clevergo...I worked more than 4+ months 24/24 I leave from university, and they just copy code change some variable names, without even notice my name anywhere... anyway  @headwindfly I will open a license issue at some time on your repo, just to know*.

**Comment by razonyang** [comment]
@kataras I hope you to read the code  of clevergo :)
I haven't seen the iris code, just found that the banner of iris is pretty, so i created one like this.
what is the reason make you think i copy your code? that make me confuse.(banner?)
of course, I very want to solve misunderstanding!

**Comment by kataras** [comment]
The sessions? are yours?

**Comment by kataras** [comment]
What do you mean just found it?

The banner and the rest of the code is there from your 'initial commit' https://github.com/headwindfly/clevergo/commit/19d877a8557f98630c8d1905be24d412904d95e1#diff-db54a02f57397abd802715262012369bR14

**Comment by razonyang** [comment]
@kataras  The rest controller inspire by beego.
http://beego.me/docs/mvc/controller/controller.md

**Comment by razonyang** [comment]
@kataras 
the original version of clevergo is here https://github.com/clevergo/clevergo, it based on net/http.
and i found the fasthttp package, and then rewrite it for fasthttp.
I thought my middleware , Context,csrf, jwt, session, captcha,etc is difference with your framework.
And it based on fasthttprouter and httprouter.
If you think I copied your code, please point out them. Thanks.
Anyway, Please do not post comments that lack of truth.

**Comment by kirillDanshin** [comment]
@kataras to be clear you need to remember that you've lied some time ago when you edited users' comments, issues etc. so if you think that `we are all friends` why you blame people?

**Comment by kataras** [comment]
@kirillDanshin you're on of Iris' fans, happy to see you here,  your feature request is  this provider: https://github.com/kataras/iris/issues/199
I didn't edit your comment, 'Bountysource  ' github integration did that...anyway

**Comment by ghost** [comment]
@kirillDanshin you're on of Iris' fans, happy to see you here,  your feature request is  this provider: https://github.com/kataras/iris/issues/199
I didn't edit your comment, 'Bountysource  ' github integration did that (which I removed before some days) ...anyway

**Comment by kirillDanshin** [comment]
@kataras no, I'm not about my own comments. I talking about other Iris' fans'/users' comments. about your committing policy: look, you merging PRs that way so nobody other can't be shown in [contributors page](https://github.com/kataras/iris/graphs/contributors). I agree that you've made huge work on creating cool framework but I can't understand your [contributing policy](https://github.com/kataras/iris/pulls?q=is%3Apr+is%3Aclosed), [vendoring policy](https://github.com/kataras/iris/issues/15), [existence of «propaganda» label](https://github.com/kataras/iris/labels) and [your issue editing](https://github.com/kataras/iris/issues/219).

**Comment by kataras** [comment]
@kirillDanshin First of all, the only think which 'touches' me is that about 'PRs' did you see any code of these exists in the iris repository now? these are from iris version 1. Two people made a pr of 3-4 lines and one function on the tests, all of this code are not exists  on Iris now, these are not shown anymore because I had to reset the repository (because of glide and 33MB iris repository length, which is some killobytes now). Secondly, about the 'vendoring policy' read the LICENSE file before posting anything like this, (this guy didn't added me on his repositories but I added him later) . Third, the 'issue editing' he talks about my parents, what did you expected from me to do?. I can't aswer more clearly, and I will not, this is not the place 'hatings' because you're author of other web framework
